### PR TITLE
Export both CJS and ESM for node

### PR DESCRIPTION
**Pull request by monsterbitar**
See https://github.com/heineiuo/isomorphic-ws/issues/30#issuecomment-1594083966 for details.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,8 +3,12 @@
   "version": "5.0.0",
   "description": "Isomorphic implementation of WebSocket",
   "main": "node.js",
+  "exports": {
+    "browser": "./browser.js",
+    "import": "./node.mjs",
+    "require": "./node.js"
+  },
   "types": "index.d.ts",
-  "browser": "browser.js",
```
Comment (piranna) [inline package.json:7]: Please left the `browser` field to keep backwards compatibility, it's not standard but some old tools could still be using it.

**Review by piranna — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (monsterbitar) [reply package.json:7]: I'm ok with this if that is what the package maintainer wants.

If they do, this PR has "allow edits by maintainers" enabled and should be an easy fix.

**Comment by JoCat** [comment]
LGTM
I did the same thing earlier in [my version of the package](https://github.com/JoCat/modern-isomorphic-ws) and as far as I know everything works correctly

**Comment by monsterbitar** [comment]
For reference, I understand why this is a difficult thing to get right - after forking and maintaining this change I've had numerous situation where this doesn't work.

I don't expect this to get merged, and have since opted to ditch CJS support entirely in my own local instance.

Feel free to re-open or use what makes sense if focus is turned back to this later.

**Comment by jeswr** [comment]
> For reference, I understand why this is a difficult thing to get right - after forking and maintaining this change I've had numerous situation where this doesn't work.

@monsterbitar is this still the case if you keep original `browser` entry as suggested in https://github.com/heineiuo/isomorphic-ws/pull/37#discussion_r1231775374.

If not would it be possible for this work to be released with that changed?

**Comment by monsterbitar** [comment]
> > For reference, I understand why this is a difficult thing to get right - after forking and maintaining this change I've had numerous situation where this doesn't work.
> 
> @monsterbitar is this still the case if you keep original `browser` entry as suggested in [#37 (comment)](https://github.com/heineiuo/isomorphic-ws/pull/37#discussion_r1231775374).
> 
> If not would it be possible for this work to be released with that changed?

yes, the issue is much deeper and there's just too many different variations and packaging setups that are close-but-not-quite-the-same across the board.

Since browsers have supported ESM for quite some time now though, I'm simply looking forward to when we drop CJS altogether and remove the additional complexity.

**Comment by piranna** [comment]
I have migrated to unws package, that's similar to this one but with full esm support.

**Pull request closed without merging**